### PR TITLE
feat(bench): add `--no-run` flag

### PR DESCRIPTION
**Pull request by GJZwiers**
This PR adds a flag to `deno bench` to cache and type-check specifiers only, similar to `deno test --no-run`

Closes https://github.com/denoland/deno/issues/18429

### CODE REVIEW

File: cli/tests/testdata/bench/no_run.ts [hunk 1]
```diff
@@ -0,0 +1 @@
+const _value: string = 1;
```
Comment (aapoalas) [inline cli/tests/testdata/bench/no_run.ts:1]: Maybe add a console.log to the file to confirm that the code is not run at all, and not just that the benchmark calls themselves are not run.

**Review by aapoalas — CHANGES_REQUESTED** [review]
Nice feature addition, thank you very much for doing this and in such short a time! I left a minor issue to use the test to verify that the code in general isn't run.

**Review by aapoalas — APPROVED** [review]
Looks good to me. Lint failed from lines that didn't change from previous... I wonder how that happens?

**Comment by GJZwiers** [comment]
Is it a breaking change in clap v4?

File: cli/args/flags.rs [hunk 2]
```diff
@@ -804,6 +805,12 @@ fn bench_subcommand() -> Command {
         .value_parser(value_parser!(PathBuf))
         .action(ArgAction::Append),
     )
+    .arg(
+      Arg::new("no-run")
+        .long("no-run")
+        .help("Cache bench modules, but don't run benchmarks")
+        .takes_value(false),
```
Comment (crowlKats) [inline cli/args/flags.rs:812]:
```suggestion
        .action(ArgAction::SetTrue),
```

File: cli/args/flags.rs [hunk 3]
```diff
@@ -2368,11 +2375,14 @@ fn bench_parse(flags: &mut Flags, matches: &mut ArgMatches) {
     Vec::new()
   };
 
+  let no_run = matches.is_present("no-run");
```
Comment (crowlKats) [inline cli/args/flags.rs:2378]:
```suggestion
  let no_run = matches.get_flag("no-run");
```

**Review by crowlKats — COMMENTED** [review]
clap v4 had a few changes:

**Pull request merged**